File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    p1 = [1, 2, 3, 4, 5]
    p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    r1, r2, r3 = len(answers)//len(p1)+1, len(answers)//len(p2)+1, len(answers)//len(p3)+1
    p1, p2, p3 = p1 * r1, p2 * r2, p3 * r3
    
    def check(a):
        cnt = 0
        for i in range(len(answers)):
            if answers[i] == a[i]:
                cnt += 1
        return cnt
    
    n1, n2, n3 = check(p1), check(p2), check(p3)
    max_x = max(n1, n2, n3)
    if n1 == max_x:
        answer.append(1)
    if n2 == max_x:
        answer.append(2)
    if n3 == max_x:
        answer.append(3)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int np1 = 5, np2 = 8, np3 = 10;
        int[] cnt = new int[3];
        int[] p1 = {1, 2, 3, 4, 5};
        int[] p2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] p3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        for(int i = 0; i < answers.length; i++){
            if(answers[i] == p1[i%np1]){
                cnt[0]++;
            }if(answers[i] == p2[i%np2]){
                cnt[1]++;
            }if(answers[i] == p3[i%np3]){
                cnt[2]++;
            }
        }
        int max = Math.max(cnt[0], Math.max(cnt[1], cnt[2]));
        ArrayList<Integer> ans = new ArrayList<>();
        for(int i = 0; i < 3; i++){
            if(max == cnt[i]){
                ans.add(i+1);
            }
        }
        return ans.stream().mapToInt(i -> i).toArray();
    }
}